app/questionnaire/create_questionnaire_manager.py

```diff
@@ -0,0 +1,52 @@
+from flask_login import current_user
+from app.validation.validator import Validator
+from app.routing.routing_engine import RoutingEngine
+from app.navigation.navigator import Navigator
+from app.questionnaire.questionnaire_manager import QuestionnaireManager
+from app.main import errors
+from app.utilities.factory import factory
+import logging
+from app.main.views.root import _load_and_parse_schema
+
+logger = logging.getLogger(__name__)
+
+
+def create_questionnaire_manager():
+
+    eq_id = current_user.get_eq_id()
+    form_type = current_user.get_form_type()
+    logger.debug("Requested questionnaire %s for form type %s", eq_id, form_type)
+
+    schema = _load_and_parse_schema()
+    if not schema:
```

Comment:
This won't work with the new method as schema can never be none. Now you've refactored, this doesn't make sense. I think it should throw an exception if schema is none.